core/src/main/java/com/confighub/core/store/Store.java

```diff
@@ -2505,30 +2505,70 @@ public Pair<PropertyKey, Collection<Property>> getPropertiesForKey( final Reposi
                                                                         String key )
           throws ConfigException
     {
-        AuditReader reader = AuditReaderFactory.get( em );
-        Number rev = reader.getRevisionNumberForDate( null == date ? new Date() : date );
-
-        AuditQuery kq = reader.createQuery().forEntitiesAtRevision( PropertyKey.class, rev );
-        kq.add( AuditEntity.property( "repository" ).eq( repository ) );
-        kq.add( AuditEntity.property( "key" ).eq( key ) );
+        PropertyKey propertyKey = null;
+        Collection<Property> properties = new ArrayList<>();
 
-        PropertyKey propertyKey;
-        try
+        if ( null == date )
         {
-            propertyKey = (PropertyKey) kq.getSingleResult();
+            try
+            {
+                propertyKey = (PropertyKey) em.createNamedQuery( "Key.getByKey" )
+                                              .setLockMode( LockModeType.NONE )
+                                              .setParameter( "key", key.toUpperCase() )
```

Comment:
This follows the existing convention. The string passed here needs to be uppercase, because the underlying query does:

`SELECT k FROM PropertyKey k WHERE repository=:repository AND UPPER(key)=:key`